specification.markdown

```diff
@@ -173,21 +173,65 @@ An EBML Stream is a file that consists of one or many EBML Documents that are co
+
+#### EBML Schema Element Attributes
+
+Within an EBML Schema the `<element>` uses the following attributes to define an EBML Element.
+
+| attribute name | required | definition |
+|----------------|----------|------------|
+| name           | Yes      | The official human-readable name of the EBML Element. The value of the name MUST be in the form of an NCName as defined by the [XML Schema specification](http://www.w3.org/TR/1999/REC-xml-names-19990114/#ns-decl). |
+| level          | Yes      | The level notes at what hierarchical depth the EBML Element may occur within an EBML Document. The initial EBML Element of an EBML Document is at level 0 and the Elements that it may contain are at level 1. The level MUST be expressed as an integer; however, the integer may be followed by a '+' symbol to indicate that the EBML Element is valid at any higher level.  |
+| global         | No       | A boolean to express if an EBML Element MUST occur at its defined level or may occur within any Parent EBML Element. |
+| id             | Yes      | The Element ID expressed in hexadecimal notation prefixed by a '0x'. |
+| mandatory      | No       | A boolean to express if the EBML Element MUST occur if the Parent EBML Element is used. If the mandatory attribute is not expressed for that Element then that element is to be considered not mandatory. |
+| multiple       | No       | A boolean to express if the EBML Element may occur within its Parent EBML Element more than once. If the multiple attribute is false or the  multiple attribute is not used to define the Element then that EBML Element MUST not occur more than once with that Element's Parent EBML Element. |
+| range          | No       | For Elements which are of numerical types (Unsigned Integer, Signed Integer, Float, and Date) a numerical range may be specified. If specified that the value of the EBML Element MUST be within the defined range inclusively. See the [section of Expressions of range](#expression-of-range) for rules applied to expression of range values. |
+| default        | No       | A default value may be provided. If an Element is mandatory but not written within its Parent EBML Element, then the reader of EBML Document MUST consider the defined default value of the Element. EBML Elements that are Master-elements MUST NOT declare a default value. |
```

Comment:
How does using the `default` attribute work for things like `DisplayWidth`/`DisplayHeight`/`BlockDuration`? They have a default value that depends on another element. How should that be encoded? Should whitespace be ignored (`OutputSamplingFrequency`'s default value has a space in it: `Sampling Frequency`)?

How are default values supposed to be encoded? Right now strings are encoded as `default="eng"`, which means that there is room for conflict/ambiguity if they have a default value that matches an element's name (i.e. `default="English"`: does it refer to the string `"English"` or the element named `English` (assuming such an element exists)).

Also, for the record, default values that depend on far away elements are a bit annoying/painful (`BlockDuration`, specifically). The default value for `BlockDuration` is not a sibling/parent/grandparent. It's a second-cousin, which makes designing a parser more difficult as it now needs access to an otherwise unrelated, far away element. I personally would like to see more restrictions on defaults that only permit them to depend on sibling/parent elements, but I realize that's an unrealistic wish. I just wanted to share a particular pain point that parser implementers will experience. (side note: `BlockDuration`'s default value still refers to the nonexistent `TrackDuration` element in `specdata.xml`)